src/Random.java

```diff
@@ -0,0 +1,192 @@
+            currentGroup += block[i] + ", ";
+            i++;
+        }
+        if (leftoverGroup != 0) {
+            randomizedGroups += "Group " + b + ": " + currentGroup;
+        }
+        /*
+         * This saves them to the textfile currentRandom, which changes everytime
+         * there's a new randomization
+         */
+        io.openWriteFile("currentRandom.txt");
+        io.writeToFile(randomizedGroups);
+        io.closeWriteFile();
+        return randomizedGroups;
+    }
+    /*
+     * In case the user wants to save the groups permanetely, this method is called
+     * on to save the file. It creates a file with a name given by the user. I had
+     * to research online on how to create a file, since the code on classroom did
+     * not teach us to do that. After the file was created, I used the changing a
+     * file code to change the contents of the file
+     */
+    public Boolean saveToFile(String filename) {
+        try {
+            File myFile = new File(filename);
+            if (myFile.createNewFile()) {
+                io.openWriteFile(filename);
+                io.writeToFile(randomizedGroups);
+                io.closeWriteFile();
+                return false;
+                // false is needed so that app.java knows if saveToFile should be rerun (if the
+                // name the user has chosen has already been used)
+            } else {
+                System.out.println("This file already exists, please choose a new name.");
+                return true;
+            }
+        } catch (IOException e) {
+            System.out.println("Error");
+            e.printStackTrace();
+            return true;
+        }
+    }
+
+    public void editArray(String names, String filename){
+    this.fileName = filename; //sets the variable fileName in this class as the String given my app.Java
+    String currentName; 
+    min = 0;
+    max = 0;
+    while (namesLeft == true) { //namesLeft is if there are still names the user has input that haven't been added to the array Block 
+        currentName = "";
+        max = names.indexOf("//", min); // max has the same index as when // is typed, so that the program gets the substring between min and max (just the name)
+        if (max >= 0) { //if the indexOf max isn't -1 (meaning there is no // in the program, it can set the names to Block)
+        currentName = names.substring(min, max); 
+        block[numStudents] = currentName;
+        numStudents++;
+        min = max + 2; //because // is two characters long, +2 must be added so the next substring doesn't include //
+        }
+        else {
+        //once max is equal to -1, namesLeft is set to false (because there are none left) and the loop ends, as all the names in the String have been given
+        namesLeft = false;
+        }
+    }
+    editFile(fileName); //calls the method editFile
+    }
+
+    public void editFile(String filename){
```

Comment:
this is I/O, it should be managed either in the controller or InputOutput (or a mix of the two)